capirca/lib/cisco.py

```diff
@@ -398,6 +398,9 @@ def __str__(self):
         and ('tcp-established' in opts or 'established' in opts)):
       if 'established' not in self.options:
         self.options.append('established')
+    if ('ip' in protocol) and ('fragments' in opts):
```

Comment:
I see you are checking on if 'ip' is in protocol here. Is this needed? Other protocols can be fragmented, yes?

Reply:
If I'm not remembering wrong (and also reviewing the code), indeed, that 'if' is making sure that fragments are only used with 'ip' protocol. Maybe there are other protocols supporting fragmentation (or something similar), but this is made here to avoid a common confusion of using fragmentation with TCP or UDP protocols.

Reply:
This likely won't work for other platforms, ciscoxr relies on ipv4 being in the protocol. What confusion are you trying to avoid with TCP and UDP with regards to fragments? Is this that you cannot use layer 4 information in the rule when you have fragments enabled?

If so I think the usage of ip and ipv6 in this area is a bit confusing considering the project as a whole.
https://github.com/google/capirca/blob/master/capirca/lib/aclgenerator.py#L92
https://github.com/google/capirca/blob/master/capirca/lib/cisco.py#L344

The use of protocol here (in cisco.py in general, not your CL specifically) is incorrect I believe. I ran into other issues with regards to how protocols are used in Capirca in a few CLs. So maybe this should be rethought.

Reply:
Sorry, this was submitted some time ago and I didn't remember clearly. But, reviewing the code, this part is just to add the "fragments" keyword. "Term.options" should contain any option defined in the policy, like "log", so this is the way to add "fragments". This control wouldn't be necessary if when we reach this point, we are 100% sure that options are set correctly, so for example, fragments are not used with other protocol than "ip", but I think we can't be 100% sure, so it's better double check that the ACL is going to be built it makes sense.